src/subscription-item.js

```diff
@@ -55,8 +57,8 @@ export default class SubscriptionItem {
     if (event && typeof event !== 'string') {
       throw new Error('Event should be string');
     }
-    if (!listener || typeof listener !== 'function') {
-      throw new Error('Listener should be function');
+    if (!listener || (typeof listener !== 'function' && !(listener instanceof Channel))) {
```

Comment:
lets add static `Channel.isValidListener()` to avoid repeating `typeof listener !== 'function' && !(listener instanceof Channel)`